File: bootcamp-2023/integration-testing/exercise/kotlin/src/test/kotlin/org/craftedsw/tripservicekata/trip/TripServiceTest.kt
import org.craftedsw.tripservicekata.trip.Trip
import org.craftedsw.tripservicekata.trip.TripService
import org.craftedsw.tripservicekata.user.User
import org.craftedsw.tripservicekata.user.UserSession

class TripServiceTest {
    @org.junit.Test
    fun foo() {
        val user = User()
        val friend = User()
        friend.addFriend(user)
        user.addFriend(friend)
        friend.addTrip(Trip("paris"))
        val tripService = TripService(UserSession(user))

        assert(tripService.getTripsByUser(friend)[0].placeName == "paris")
    }
}

File: bootcamp-2023/integration-testing/exercise/php/test/Test/TripServiceKata/Trip/TripServiceTest.php
<?php

namespace Test\TripServiceKata\Trip;

use PHPUnit\Framework\TestCase;
use TripServiceKata\Trip\TripService;
use TripServiceKata\User\User;
use TripServiceKata\User\UserSession;

class TripServiceTest extends TestCase
{
    /** @test */
    public function it_does_something()
    {
        $user = new User();
        $friend = new User();
        $user->addFriend($friend);
        $friend->addFriend($user);
        $friend->addTrip('paris');
        $service = new TripService(new UserSession($user));

        $this->assertTrue($service->getTripsByUser($friend) == ['paris']);
    }
}
